nifpga/session.py

```diff
@@ -296,11 +297,24 @@ class _Register(object):
     of this class.
 
     """
-    def __init__(self, session, nifpga, bitfile_register, base_address_on_device):
+    def __init__(self,
+                 session,
+                 nifpga,
+                 bitfile_register,
+                 base_address_on_device,
+                 read_func=None,
+                 write_func=None):
         self._datatype = bitfile_register.datatype
         self._name = bitfile_register.name
         self._session = session
-        self._nifpga = nifpga
+        if read_func is None:
+            self._read_func = nifpga["Read%s" % self.datatype]
```

Comment:
No, in the case of Fixed point ReadFxp does not exist, that is why there is the option to pass in a read_function into register like I did for this case.

Reply:
I intended my question to be more like: "How do we know that all callers will provide a function when the nifpga lookup would fail? And maybe also, how do we know that all callers will provide None when the nifpga lookup will succeed?"

I guess we expect that this particular method can only be called by your code (users are not supposed to instantiate this class), so we "only" need to trust your calling code to be correct. I suppose a corollary assumption is that no unqualified person will touch your code later.

I think I've seen similar lookups elsewhere in your changes. Are you confident that no user code can trick you into performing an invalid lookup?